### PR TITLE
Isomapper patch

### PR DESCRIPTION
**Pull request by valasatava**
isoform to canonical sequence mapping bag fix

### CODE REVIEW

File: src/main/java/org/rcsb/uniprot/isoform/IsoformMapper.java [hunk 1]
```diff
@@ -37,62 +38,75 @@
...
+        // handle heads and tails that are not included in the alignment
+        if ( ! ArrayUtils.contains(s1.getSequenceFromAlignment(), coordinate) ) {
+            return -1;
+        }
+
         try {
-            int aligPos = s1.getAlignmentIndexAt(coordinate);
-            return s2.getSequenceIndexAt(aligPos);
+
+            int alignmentInd = ArrayUtils.indexOf(s1.getSequenceFromAlignment(), coordinate);
```
Comment (andreasprlic) [inline src/main/java/org/rcsb/uniprot/isoform/IsoformMapper.java]: I think you can get the alignment index position also from the pair

Comment (valasatava) [inline src/main/java/org/rcsb/uniprot/isoform/IsoformMapper.java]: I can see how it's possible to get a position in the sequence given the alignment index. I need the opposite - given the sequence position, what is the index of that position in the alignment. Could you, please, elaborate on your comment?

**Comment by andreasprlic** [comment]
Nevermind , your approach is right. That api method does not exist...



-- 
-----------------------------------------------------------------------
Dr. Andreas Prlic
RCSB PDB Protein Data Bank
Technical & Scientific Team Lead
University of California, San Diego

Editor Software Section
PLOS Computational Biology

BioJava Project Lead
-----------------------------------------------------------------------

**Comment by valasatava** [comment]
Oh, I think I figured out what you meant…I will try now to make a better use of existing APIs.

Yana

**Pull request merged**